src/pynwb/ndx_patterned_ogen/patterned_ogen.py

```diff
@@ -524,3 +614,54 @@ def add_interval(self, **kwargs):
         Add a stimulation parameters for a specific stimulus onset.
         """
         super(PatternedOptogeneticStimulusTable, self).add_interval(**kwargs)
+
+        if isinstance(kwargs["power"], (list, np.ndarray, tuple)):
+            raise ValueError(
+                "'power' should be defined as scalar. Use 'power_per_rois' to store photostimulation at different"
+                " powers, for each rois in target."
+            )
+        if isinstance(kwargs["frequency"], (list, np.ndarray, tuple)):
+            raise ValueError(
+                "'frequency' should be defined as scalar. Use 'frequency_per_rois' to store photostimulation at"
+                " different frequency, for each rois in target."
+            )
+
+        if isinstance(kwargs["pulse_width"], (list, np.ndarray, tuple)):
+            raise ValueError(
+                "'pulse_width' should be defined as scalar. Use 'pulse_width_per_rois' to store photostimulation with"
+                " different pulse width, for each rois in target."
+            )
+
+        n_targets = len(kwargs["targets"].targeted_rois[:])
+
+        if kwargs["power_per_rois"] is not None:
+            n_elements = len(kwargs["power_per_rois"])
+            if n_elements != n_targets:
+                raise ValueError(
+                    f"'power_per_rois' has {n_elements} elements but it must have"
+                    f" {n_targets} elements as 'targeted_roi'."
```

Comment:
```suggestion
                    f" {n_targets} elements to match the length of 'targeted_roi'."
```

More readable IMO, for an example,

```python
'power_per_rois' has 5 elements but it must have 15 elements to match the length of 'targeted_roi'.
```